packages/service/src/plugin.ts

```diff
@@ -3,20 +3,16 @@ import type { Logger } from 'winston';
 
 import type { RehearsalService } from './rehearsal-service';
 
-export class Plugin {
-  protected readonly logger?: Logger;
-  protected readonly reporter: Reporter;
-  protected readonly service: RehearsalService;
+export interface Plugin<PluginOptions> {
+  run(fileName: string, options: PluginOptions): PluginResult;
+}
 
-  constructor(service: RehearsalService, reporter: Reporter, logger?: Logger) {
-    this.service = service;
-    this.logger = logger;
-    this.reporter = reporter;
-  }
+export interface PluginOptions {
+  [key: string]: unknown;
 
```

Comment:
Can we add basePath to PluginOptions?

Reply:
You can add this option to specific plugin, like for Lint here:
https://github.com/rehearsal-js/rehearsal-js/pull/675/files#diff-3be91568bd3d2f0e95e214fa7b01784fa345decb2da97e9211aaecc92b30bda7R7-R9

but I will think about sending PluginContext (cli, basePath, service, etc) together with PluginOptions next PR.